### PR TITLE
fix(database-chabge): allow delete finished schedule

### PR DESCRIPTION
**Pull request by kiko-art**
#### What type of PR is this?
type-bug

#### What this PR does / why we need it:
allow delete finished schedule
#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by yhilmare — DISMISSED** [review]
lgtm

File: server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/ScheduleService.java [hunk 1]
```diff
@@ -303,9 +303,11 @@ public ChangeScheduleResp changeSchedule(ScheduleChangeParams req) {
                 throw new IllegalStateException("Update schedule is not allowed.");
             }
             if (req.getOperationType() == OperationType.DELETE
-                    && targetSchedule.getStatus() != ScheduleStatus.TERMINATED) {
+                    && (targetSchedule.getStatus() != ScheduleStatus.TERMINATED
+                            || targetSchedule.getStatus() != ScheduleStatus.COMPLETED)) {
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/ScheduleService.java:307]: the judgement here is not true

Comment (kiko-art) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/ScheduleService.java]: Already changed.

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**